content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,460 @@
+```
+
+In the first two lines, we provide the paths to the Unikraft core repository and the directory where we can put additional libraries.
+Since the `helloworld` application uses [`nolibc`](https://github.com/unikraft/unikraft/tree/staging/lib/nolibc) as the C library, which comes with the Unikraft core, we do not have any external dependencies.
+We thus do not have to download or reference any libraries in the `Makefile` and can leave the `LIBS` variable empty.
+
+When the application becomes more complex, it is likely that not all functionality is included in the Unikraft core repository.
+As you might have noticed already from the code organization in the core repository (`lib/*`), Unikraft puts an emphasis on encapsulating functionality in libraries that can be easily added, removed, or replaced to tailor the unikernel to the application's needs.
+
+To add a library, clone it into the `libs/` directory created earlier.
+In this example, we add the `lightweight IP` (`lwip`) network stack [`lib-lwip`](https://github.com/unikraft/lib-lwip):
+
+```console
+$ cd ../../libs/
+$ git clone https://github.com/unikraft/lib-lwip
+```
+
+Next, we reference the library in the `Makefile` of `app-helloworld`, making it available for selection in the configuration:
+
+```Makefile
+UK_LIBS ?= $(PWD)/../../libs
+LIBS := $(UK_LIBS)/lib-lwip
+```
+
+We can now re-run
+
+```console
+$ make menuconfig
+```
+
+and select `lwip` in the `Library Configuration` sub-menu.
+We can then rebuild the unikernel:
+
+```console
+$ make clean
+$ make prepare
+$ make
+```
+
+The `prepare` step gives the build system the chance to pull the actual source code for `lwip` and apply necessary patches to make it work with Unikraft.
+This has to be done only once after adding a new library.
+
+### Add New Source Files to the Build
+
+To add new source files to the build, we add them to the `Makefile.uk` file.
+We can choose to add source files only if certain configuration options are set.
+
+```makefile
+$(eval $(call addlib,apphelloworld))
+
+APPHELLOWORLD_SRCS-y += $(APPHELLOWORLD_BASE)/main.c
+APPHELLOWORLD_SRCS-$(CONFIG_APPHELLOWORLD_LOGGER) += $(APPHELLOWORLD_BASE)/logger.c
+```
+
+This will add the `logger.c` source file to the build system only if the `CONFIG_APPHELLOWORLD_LOGGER` option is enabled in the configuration step.
+We will discuss how to add the new configuration option below.
+
+### Use Different Configuration Options
+
+We can edit the `Config.uk` file to add additional dependencies and configuration options.
+
+```kconfig
+### Invisible option for dependencies
+config APPHELLOWORLD_DEPENDENCIES
+        bool
+        default y
+        select LIBMUSL
+        select LIBLWIP
+
+config APPHELLOWORLD_LOGGER
+        bool "Add the logger to the build system"
+        default n
+```
+
+This will select the `musl` and `lwip` dependencies and create a new configuration option that can be set in the `Application Options` screen in the `menuconfig`.
```

Comment:
```suggestion
This will automatically select the `KVM` platform and the `lwip` external library and create a new configuration option that can be set in the `Application Options` screen in the `menuconfig`.
```